nrfx/drivers/src/nrfx_lpcomp.c

```diff
@@ -183,6 +183,34 @@ nrfx_err_t nrfx_lpcomp_init(nrfx_lpcomp_config_t const * p_config,
     return err_code;
 }
 
+nrfx_err_t nrfx_lpcomp_reconfigure(nrfx_lpcomp_config_t const * p_config)
+{
+    NRFX_ASSERT(p_config);
+    nrfx_err_t err_code;
+
+    if (m_state == NRFX_DRV_STATE_UNINITIALIZED)
+    {
+        err_code = NRFX_ERROR_INVALID_STATE;
+        NRFX_LOG_WARNING("Function: %s, error code: %s.",
+                         __func__,
+                         NRFX_LOG_ERROR_STRING_GET(err_code));
+        return err_code;
+
+    }
+    else if (m_state == NRFX_DRV_STATE_POWERED_ON)
+    {
+        err_code = NRFX_ERROR_BUSY;
+        NRFX_LOG_WARNING("Function: %s, error code: %s.",
+                         __func__,
+                         NRFX_LOG_ERROR_STRING_GET(err_code));
+        return err_code;
+    }
```

Comment:
nit, I would expect here something like
```suggestion
    } else if (m_state == expected state or states) {
      ...
      return NRFX_SUCCESS;
    } else {
      NRFX_ASSERT(false);
      return NRFX_ERROR_...;
    }
```

Reply:
The only possible state left is `NRFX_DRV_STATE_INITIALIZED`, which is the expected one, so I believe that further if statements can be avoided

Reply:
The further `if` catches unexpected errors when `m_state` gets unexpected value. E.g. if other module writes over this module's memory or when new states are added during refactoring and someone forgets to update this function.

It also allows someone reading the code to not check the `m_state` enumeration, because all states are clearly captured in this function.

BTW, probably `switch-case` would be even clearer to read

But I agree, it's not a big issue. More like a safer programming style.

Reply:
Good point